### PR TITLE
First merge into master

### PR DESCRIPTION
**Pull request by andre-lima**
This includes an initial webpack configuration, React, eslint, tests with mocha+chai+enzyme, a global controller file to be used for talking to the server.

Some lints are failing due mostly to unused variables.

Some of the structure (config folder, schema files and controller) may change as I try to talk to the server and learn how not to do somethings.

Known issue: Jsx files are not being linted.

### CODE REVIEW

File: .babelrc [hunk 1]
```diff
@@ -0,0 +1 @@
+{ "presets": ["es2015", "stage-2", "react"] }
```
Comment (peakwinter) [inline .babelrc]: Kind of a question of preference, but you can also put your Babel config as a section of your `package.json`, to cut down on extraneous files :)

File: config/config.js [hunk 2]
```diff
@@ -0,0 +1,23 @@
+import schemas from './schemas/schemas';
+
+const url = 'http://localhost:4040';
+
+const endpoints = {
+  entities: { url: `${url}/api/entities`, schema: schemas.task },
+  task: { url: `${url}/api/task`, schema: schemas.task },
+  tasklist: { url: `${url}/api/tasklist`, schema: schemas.tasklist },
+  taskboard: { url: `${url}/api/taskboard`, schema: schemas.taskboard },
```
Comment (peakwinter) [inline config/config.js]: The schemas here will only be used server-side for validation. You can still have them in your config if you'd like, but you won't be able to use an automatic validation system like `ajv` because things like unique IDs are given out by the server and not assigned by the front end.

File: config/schemas/task.json [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+      "$ref": "definitions/uuid.json",
+      "description": "The UUID of a Sylow entity that is tasked with completing this task, if necessary."
+    },
+    "meta": {
+      "type": "object",
+      "description": "Additional metadata to provide along with the document.",
+      "additionalProperties": true
+    }
+  }
+}
```
Comment (peakwinter) [inline config/schemas/task.json]: If you'd still like to use the schemas client-side, you should link them from the schema server (e.g. `https://sylow.network/schemas/v1/base.json`) instead, so that the proper version can be authoritatively sourced.

File: config/test_helper.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import jsdom from 'jsdom';
+import Controller from '../src/controller/app_controller';
+
+const { JSDOM } = jsdom;
+const dom = new JSDOM('<!DOCTYPE html><html><body></body></html>');
+
+global.document = dom.window.document;
+global.window = dom.window;
+
+Object.keys(global.window).forEach((key) => {
```
Comment (peakwinter) [inline config/test_helper.js]: Have you checked how this runs to make sure there aren't any extraneous things on `global` being put onto `window` needlessly?

File: src/components/LoginBox/__LoginForm/index.jsx [hunk 5]
```diff
@@ -0,0 +1,30 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+export default class extends React.Component {
+  constructor() {
```
Comment (peakwinter) [inline src/components/LoginBox/__LoginForm/index.jsx]: I don't think the `constructor` here is required :)

File: src/components/LoginBox/__LoginForm/index.jsx [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+    super();
+  }
+  static propTypes = {
+  }
+
+  // Testing a request when pushing the Login button - DELETE LATER!
+  _testRequest(endpoint) {
+    global.AppController.makeGetRequest('entities')
```
Comment (peakwinter) [inline src/components/LoginBox/__LoginForm/index.jsx]: I know we talked about this, but I can't remember the exact reason for importing the App controller in this way instead of just via an `import` ?

File: src/components/LoginBox/index.jsx [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+import PropTypes from 'prop-types';
+import LoginForm from './__LoginForm';
+import logo from './assets/logo.png';
+
+export default class LoginBox extends React.Component {
+  static propTypes = {
+
+  }
+
+  constructor(props) {
```
Comment (peakwinter) [inline src/components/LoginBox/index.jsx]: Same here, is the `constructor` required in this context?

File: src/components/LoginBox/__LoginForm/index.jsx (hunk 5 above)
Comment (andre-lima) [inline src/components/LoginBox/__LoginForm/index.jsx]: Some constructors an propTypes were left as placeholders, since I was going to start adding more logic soon, which didn't happen.
I'll remove them to clean things up for now.

File: src/components/LoginBox/index.jsx (hunk 7 above)
Comment (andre-lima) [inline src/components/LoginBox/index.jsx]: Same...

File: src/components/LoginBox/__LoginForm/index.jsx (hunk 6 above)
Comment (andre-lima) [inline src/components/LoginBox/__LoginForm/index.jsx]: It allows for testing of the controller inside the React components. I don't remember exactly what happened, but it just wouldn't work by importing the controller.
I actually think that using mocha for testing React is not a very good idea, as it requires Enzyme and JSDOM for a lot of things, and a minor unrelated change can break the tests.

File: config/schemas/task.json (hunk 3 above)
Comment (andre-lima) [inline config/schemas/task.json]: Thanks.

File: .babelrc (hunk 1 above)
Comment (andre-lima) [inline .babelrc]: Done.

File: config/test_helper.js (hunk 4 above)
Comment (andre-lima) [inline config/test_helper.js]: This is only run in node, during tests. 'window' is a fake jsdom window object.

Bur here's the list:

_eventHandlers 
_core 
_globalProxy 
__timers 
_document 
_sessionHistory 
_currentSessionHistoryEntryIndex 
_virtualConsole 
_runScripts 
_top 
_parent 
_frameElement 
_length 
length 
frameElement 
frames 
self 
parent 
top 
location 
history 
navigator 
addEventListener 
removeEventListener 
dispatchEvent 
__stopAllTimers 
postMessage 
atob 
btoa 
FileReader 
XMLHttpRequest 
stop 
close 
getComputedStyle 
name 
innerWidth 
innerHeight 
outerWidth 
outerHeight 
pageXOffset 
pageYOffset 
screenX 
screenY 
screenLeft 
screenTop 
scrollX 
scrollY 
scrollTop 
scrollLeft 
screen 
alert 
blur 
confirm 
createPopup 
focus 
moveBy 
moveTo 
open 
print 
prompt 
resizeBy 
resizeTo 
scroll 
scrollBy 
scrollTo 
HTMLElement

**Comment by andre-lima** [comment]
Changes made. Please check.
Thanks.
